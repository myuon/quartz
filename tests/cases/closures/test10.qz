fun f(a: i32, p: fun[(i32), nil]) {
    println("{}", a.to_string());
    p(7);
}

fun main() {
    let a = 10;

    f(
        5,
        fun (resp_id: i32) {
            f(
                a,
                fun (arg: i32) {
                    println("called with {}", arg.to_string());
                },
            );
        },
    );

    println("ok");
}

