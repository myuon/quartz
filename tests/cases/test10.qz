fun main(): i32 {
    let p = make[ptr[i32]](20);
    p.(0) = 10;
    p.(1) = 20;
    p.(2) = p.(0) + p.(1);

    return p.(2);
}

