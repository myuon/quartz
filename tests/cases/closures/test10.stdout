5
10
called with 7
ok
